environs/bootstrap/bootstrap.go

```diff
@@ -218,8 +206,10 @@ func Bootstrap(ctx environs.BootstrapContext, environ environs.Environ, args Boo
 	// If no arch is specified as a constraint, we'll bootstrap
 	// on the same arch as the client used to bootstrap.
 	var bootstrapArch string
-	if bootstrapConstraints.Arch != nil {
-		bootstrapArch = *bootstrapConstraints.Arch
+	if args.BootstrapConstraints.Arch != nil {
```

Comment:
They are not merged here yet. They are merged in juju/cmd/commands/bootstrap. But if this method is not called from bootstrap cli, they still need to be merged. Validator that does it properly as well as validates them is only available later on. Hence, this logic here to ensure that we are getting correct image metadata.